src/components/uielements/assets/assetIcon/AssetIcon.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState, useMemo, useEffect, useCallback } from 'react'
+
+import { Asset, AssetTicker } from '@thorchain/asgardex-util'
+
+import bnbIcon from '../../../../assets/svg/coin-bnb.svg'
+import runeIcon from '../../../../assets/svg/rune-flash-icon.svg'
+import { getIntFromName, rainbowStop } from '../../../../helpers/colorHelpers'
+import * as Styled from './AssetIcon.style'
+import { Size } from './types'
+
+interface Props extends React.HTMLAttributes<HTMLDivElement> {
+  size?: Size
+  asset: Asset
+}
+
+const AssetIcon: React.FC<Props> = (props: Props): JSX.Element => {
+  const { asset, size = 'normal' } = props
+  const [error, setError] = useState(false)
+
+  const imgUrl = useMemo(() => {
+    const { symbol, ticker } = asset
+    const assetId = symbol
+    // currently we do load assets for Binance chain only
+    // Note: Trustwallet supports asset names for mainnet only. For testnet we will use the `IconFallback` component
+    let url = `https://raw.githubusercontent.com/trustwallet/assets/master/blockchains/binance/assets/${assetId}/logo.png`
+
+    if (ticker === AssetTicker.RUNE) {
+      // Always use "our" Rune asset
+      url = runeIcon
+    }
+    if (ticker === 'BNB') {
+      // Since BNB is blacklisted at TrustWallet's asset, we have to use "our" own BNB icon
+      // (see https://github.com/trustwallet/assets/blob/master/blockchains/binance/blacklist.json)
+      url = bnbIcon
+    }
+    return url
+  }, [asset])
+
+  const onErrorHandler = useCallback(() => {
+    setError(true)
+  }, [])
+
+  const renderIcon = useMemo(
+    () => (
+      <Styled.IconWrapper size={size}>
+        <Styled.Icon src={imgUrl} size={size} onError={onErrorHandler} />{' '}
```

Comment:
i believe we would need something like this in future when we gonna want load any image. maybe we can write some hook helper like this ?

```ts
const useRemoteImage = (url: string) => {
  const [imageRd, setImageRd] = useState<RemoteData<Error, string>>(pending);
  const image = new Image();
  image.src = url;

  image.onload = () => {
    setImageRd(success(url));
  }
  
  image.onerror = () => {
    setImageRd(failure(Error('failed to load image')))
  }

  return imageRd;
}
```

in this case we do need to resolve flags in render method and have a pending state until image loads

Reply:
Nice idea!

Reply:
^ Issue has been filed https://github.com/thorchain/asgardex-electron/issues/250